pandas/core/groupby/groupby.py

```diff
@@ -764,7 +766,21 @@ def _try_cast(self, result, obj, numeric_only=False):
             dtype = obj.dtype
 
         if not is_scalar(result):
-            if is_extension_array_dtype(dtype):
+            if is_datetime64tz_dtype(dtype):
+                # GH 23683
+                # Prior results _may_ have been generated in UTC.
+                # Ensure we localize to UTC first before converting
+                # to the target timezone
+                try:
+                    utc_dtype = DatetimeTZDtype(tz='UTC')
+                    result = obj._values._from_sequence(result,
+                                                        dtype=utc_dtype)
```

Comment:
can't you just pass ``dtype='UTC'`` instead of creating the actual dtype? (as that we rarely do for timezones)

Reply:
Just tried and passing `dtype='UTC'` didn't work here. (IMO it's more explicit too that we're passing a dtype object here)

Reply:
why doesn’t this work?  that is very odd 

we rarely actually pass the dtype object anywhere else

Reply:
maybe you actually need to construct as naive then localize? (these should be the same though)

Reply:
Some tests were failing; possibly the order of operations are different in `_from_sequence`.

Using `tz_localize('UTC').tz_convert(dtype.tz)` worked though